### ISSUE TITLE
Error in loss function

### ISSUE THREAD

**Issue by ymzhang1919** [MarvinTeichmann/tensorflow-fcn#18]
epsilon = tf.constant(value=1e-4)     
       logits = logits + epsilon
       softmax = tf.nn.softmax(logits)

It should be 
       epsilon = tf.constant(value=1e-4)    
       softmax = tf.nn.softmax(logits) + epsilon

**Comment by MarvinTeichmann** [comment]
Why? The purpose of the epsilon is to avoid numerical instability.

**Comment by ymzhang1919** [comment]
Logits are real numbers. How can you improve the numerical stability by adding a small positive number to them? On the other hand, adding a number to softmax makes the log() operation more robust.

**Comment by ymzhang1919** [comment]
I understand the purpose. But I don't think it works. Logits are real numbers (may be negative). How can you improve the numerical stability by adding a small positive number to them? 

On the other hand, adding a number to softmax makes the log() operation more robust.

**Comment by ymzhang1919** [comment]
I understand the purpose. But I don't think it works. Logits are real numbers (may be negative). How can you improve the numerical stability by adding a small positive number to them?

On the other hand, adding a small positive number to softmax makes the log() operation more robust.

**Comment by ymzhang1919** [comment]
I understand the purpose, but I don't understand how it works. Logits can be big negative numbers. How can you improve the numerical stability of the softmax() operation by adding a small positive number to logits?

On the other hand, adding a small positive number to softmax makes the log() operation more robust.

If I am wrong, can you explain it in detail? Thx.

**Comment by MarvinTeichmann** [comment]
You are right, I have fixed it.
